storage/innobase/row/row0mysql.cc

```diff
@@ -688,7 +688,7 @@ row_mysql_handle_errors(
 			break;
 		}
 		/* MariaDB will roll back the entire transaction. */
-		trx->bulk_insert = false;
+		trx->bulk_insert &= TRX_DDL_BULK;
 		trx->last_stmt_start = 0;
 		break;
```

Comment:
I think that we need a better comment here:
```c++
/* For DML or ALTER TABLE...ALGORITHM=COPY, MariaDB will roll back the entire transaction in ... (fill in).
For CREATE TABLE...SELECT, the transaction will be rolled back in ... */
```
After all, this is the core part of the bug fix.

Reply:
DML/ ALTER TABLE..COPY does partial rollback and clear bulk buffer in row_mysql_handle_errors(). 
For create..select, we do rollback the transaction in ha_innobase::delete_table()